src/components/fields/image-upload/image-manager/image-manager.ts

```diff
@@ -137,17 +137,21 @@ export const ImageManager = (props: IProps) => {
 		setErrorCount((prev) => Math.max(0, prev + updatedManagerErrorCount - managerErrorCount));
 		setManagerErrorCount(updatedManagerErrorCount);
 
-		onChange({
-			target: {
-				value: images
-					.filter(({ status }) => status === EImageStatus.UPLOADED)
-					.map(({ dataURL, drawingDataURL, name, uploadResponse }) => ({
-						fileName: name,
-						dataURL: drawingDataURL || dataURL,
-						uploadResponse,
-					})),
-			},
-		});
+		// checking for previous images to prevent firing onChange on first mount
+		// otherwise it will crash on SSR (likely due to calling validation when field is not registered)
+		if (previousImages) {
```

Comment:
this is highly dependent on the initial value of previousImages/images, but I don't see a better way. wondering if `setValue` with `shouldValidate=false` when the field is not dirty would work

Reply:
ok just gave it a try and managed to use `setValue` through a combination of `shouldValidate` and `shouldDirty`, agree that this is a better way indeed